lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -491,31 +520,47 @@ export class UserAgentApplication {
         scopes = this.filterScopes(scopes);
       }
 
-      // Extract ADAL id_token if it exists
-      let idTokenObject;
-      idTokenObject = this.extractADALIdToken();
+      // construct extraQueryParams string from the request
+      let extraQueryParameters = Utils.constructExtraQueryParametersString(request.extraQueryParameters);
+      extraQueryParameters = Utils.addPromptParameter(extraQueryParameters, request.prompt);
+      let user = this.getUser();
 
-      // silent login if ADAL id_token is retrieved successfully - SSO
-      if (idTokenObject && !scopes) {
-        this.logger.info("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
-        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(idTokenObject, extraQueryParameters);
+      // if the developer provides one of these, give preference to developer choice
+      if (request.account || request.sid || request.loginHint) {
```

Comment:
can we factor this check into something ?   requestHasHint(request)

that way if we update what that check means we only update in one place and the fn name is self documenting what the intent of the predicate is

Reply:
Optimizations in the new code.